packages/components/src/modal/Modal.tsx

```diff
@@ -0,0 +1,166 @@
+import classNames from 'classnames';
+import React, {
+  ReactElement,
+  ReactNode,
+  useCallback,
+  useEffect,
+  useRef,
+} from 'react';
+import ReactDOM from 'react-dom';
+import { CSSTransition } from 'react-transition-group';
+import ThemeExport from '../ThemeExport';
+
+interface ModalProps {
+  className?: string;
+  children?: ReactNode;
+  role?: string;
+  keyboard?: boolean;
+  isOpen?: boolean;
+  centered?: boolean;
+  size?: 'sm' | 'lg' | 'xl' | undefined;
+  onOpened?: () => void;
+  onClosed?: () => void;
+  toggle?: () => void;
+  'data-testid'?: string;
+}
+
+const Modal = ({
+  className = 'theme-bg-light',
+  children,
+  role = 'role',
+  keyboard = true,
+  isOpen = false,
+  centered = false,
+  size,
+  onOpened,
+  onClosed,
+  toggle,
+  'data-testid': dataTestId,
+}: ModalProps): ReactElement => {
+  const element = useRef<HTMLElement>();
+
+  const handleKeyDown = useCallback(
+    (event: KeyboardEvent): void => {
+      switch (event.key) {
+        case 'Escape':
+          toggle?.();
+          break;
+        default:
+          break;
+      }
+    },
+    [toggle]
+  );
+
+  useEffect(
+    function addKeydownEventListener() {
+      if (isOpen && keyboard) {
+        window.addEventListener('keydown', handleKeyDown);
+        return () => window.removeEventListener('keydown', handleKeyDown);
+      }
+    },
+    [handleKeyDown, isOpen, keyboard]
+  );
+
+  useEffect(
+    function open() {
+      if (isOpen && onOpened) {
+        onOpened();
+      }
+    },
+    [onOpened, isOpen]
+  );
+
+  useEffect(
+    function closed() {
+      if (!isOpen && onClosed) {
+        onClosed();
+      }
+    },
+    [onClosed, isOpen]
+  );
+
+  if (isOpen && !element.current) {
+    element.current = document.createElement('div');
+    element.current.setAttribute(
+      'style',
+      'padding-right: 15px; display: block'
+    );
+    element.current.setAttribute('role', 'presentation');
+    document.body.appendChild(element.current);
+  }
+
+  const onExited = () => {
+    if (element.current) {
+      document.body.removeChild(element.current);
+      element.current = undefined;
+    }
+  };
+
+  return element.current ? (
+    ReactDOM.createPortal(
+      <>
+        <CSSTransition
+          appear
+          mountOnEnter
+          unmountOnExit
+          in={isOpen}
+          classNames={{
+            enterActive: 'show',
+            enterDone: 'show',
+          }}
+          timeout={ThemeExport.transitionMs}
+          onExited={onExited}
+        >
+          <div
+            className={classNames('modal-backdrop fade')}
+            style={{ zIndex: 1050 }}
+          />
+        </CSSTransition>
+        <CSSTransition
+          appear
+          mountOnEnter
+          unmountOnExit
+          in={isOpen}
+          classNames={{
+            enterActive: 'show',
+            enterDone: 'show',
```

Comment:
Playing with it a bit more, and noticed a couple things:
- `enterActive` class doesn't seem to be getting applied at all (not sure why??)
- `enterDone` class gets applied after the `timeout` transition has completed
- `appearActive` appears to apply right away, then `appearDone` gets applied after the transition

Based on that information, you should change the dialog `classNames` to only specify `appearDone` (so the `show` class only gets applied when the transition is completed; which will _actually_ kick off the CSS transition), and change the `enterActive`/`enterDone` classNames on the background to be `appearActive` and `appearDone` instead (right now it's just delaying the transition by 150ms and then applying the backdrop. You can try with a really long timeout (like 3000) to see what I mean).
```suggestion
            appearDone: 'show',
```